src/System.Private.CoreLib/shared/System/Guid.cs

```diff
@@ -768,43 +768,51 @@ private static ReadOnlySpan<char> EatAllWhitespace(ReadOnlySpan<char> str)
             str[i] == '0' &&
             (str[i + 1] | 0x20) == 'x';
 
-        [MethodImpl(MethodImplOptions.AggressiveInlining)]
-        private void WriteByteHelper(Span<byte> destination)
-        {
-            destination[0] = (byte)(_a);
-            destination[1] = (byte)(_a >> 8);
-            destination[2] = (byte)(_a >> 16);
-            destination[3] = (byte)(_a >> 24);
-            destination[4] = (byte)(_b);
-            destination[5] = (byte)(_b >> 8);
-            destination[6] = (byte)(_c);
-            destination[7] = (byte)(_c >> 8);
-            destination[8] = _d;
-            destination[9] = _e;
-            destination[10] = _f;
-            destination[11] = _g;
-            destination[12] = _h;
-            destination[13] = _i;
-            destination[14] = _j;
-            destination[15] = _k;
-        }
-
         // Returns an unsigned byte array containing the GUID.
         public byte[] ToByteArray()
         {
             var g = new byte[16];
-            WriteByteHelper(g);
+            if (BitConverter.IsLittleEndian)
+            {
+                MemoryMarshal.TryWrite<Guid>(g, ref this);
+            }
+            else
+            {
+                TryWriteBytes(g);
+            }
             return g;
         }
 
         // Returns whether bytes are sucessfully written to given span.
         public bool TryWriteBytes(Span<byte> destination)
         {
-            if (destination.Length < 16)
-                return false;
-
-            WriteByteHelper(destination);
-            return true;
+            if (BitConverter.IsLittleEndian)
+            {
+                return MemoryMarshal.TryWrite(destination, ref this);
+            }
+
+            // slower path for BigEndian
+            if ((uint)destination.Length >= 16)
```

Comment:
Nit: Does writing it this way help anything? It is generally more readable to use early returns, like:

```
 if (destination.Length < 16)
    return false;

destination[0] = ...;
....
```